rotkehlchen/tests/unit/globaldb/test_globaldb_cache.py

```diff
@@ -171,6 +176,27 @@ def test_velodrome_cache(optimism_inquirer):
     assert all((identifier,) in asset_identifiers for identifier in VELODROME_SOME_EXPECTED_ASSETS)
 
 
+@pytest.mark.vcr()
+def test_convex_cache(ethereum_inquirer):
+    """Test convex pools querying and caching mechanism"""
+    with GlobalDBHandler().conn.write_ctx() as write_cursor:
+        # make sure that curve cache is clear at the beginning
+        write_cursor.execute('DELETE FROM general_cache WHERE key LIKE "%CONVEX%"')
```

Comment:
Okay so question here. Both here and in Curve there is this `write_cursor.execute('DELETE FROM general_cache WHERE key LIKE "%CONVEX%"')`. This deletes all cached data and does a full clean pull of all the pools from the chain.

It used to be like that for velodrome too, but I changed it here: https://github.com/rotki/rotki/pull/6788/files#diff-098f0ac6382a14207e3774683524c1ecdf2e15742e39d0925b64ca8fe2520348L154 as one of the changes I did to improve tests and speed.

Can you maybe do the same here and in the curve test below? (you would need to rewrite the cassettes for both tests).

What we would gain by not deleting and repulling all but only some is:

1. A lot less on-chain calls. Even when recorded in a cassette the more recorded calls a test has the slower it is
2. We would test the case where there is some pools in the DB, some new on-chain and we only pull thew new stuff.

To do this we would need to have a GlobalDB where some pools data **are missing**. Otherwise test would be a no-op. And you would need to test against those missing pools.

Reply:
velodrome was a no-op, now fixed, and convex too has been implemented